### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by NVB9ALT** [kolos26/GeoFS-CC_rnav#1]
Your Realism Pack fork a) doesn't have an issues tab, and b) is very out of date lol. Anyway...
I've been blocked on the wiki for 3 weeks (I think), and I do not have discord, so here I am. If you want, I can link you my Guilded server and we can talk there.
I've got one final update I want to push to main.js, then the project is all yours.

**Comment by kolos26** [comment]
I don't know this Guilded server. Github will be perfect. I have updated the fork and added a issues tab. I'm already working on the toggle options.
And this project won't be mine, it will be ours! If you have any ideas I will try to implement them. (Just as I did with the airbus sound effects.)
And I'm looking forward to the _final_ -> **next** update.

**Comment by kolos26** [comment]
I will ask a friend of mine about proxys. Can you please explain your problem, and what you need to be able to run through the proxy?

**Comment by kolos26** [comment]
And please also send me what proxy are you using!

**Comment by NVB9ALT** [comment]
This is a Replit deployment of the Ultraviolet proxy, forked from AriakimTaiyo's repl (which doesn't exist anymore). I've edited (I think) uv.handler.js, to automatically run the bookmarklet console and realism pack. https://replit.com/@CovelJulian/GeoFS-proxy
Issue: Doesn't work. There's at least four different ways it breaks, trying to load/run the repl rolls the dice on one of them.
- Repl fails to load
- Repl fails to boot
- Frontend fails to load
- Proxy fails to connect

**Comment by kolos26** [comment]
For me it says connection is not secure. But for you it must brake sometime later.
When was the last time it was running?
Were there any browser updates after that,? 
Last known (= as I noticed ) GeoFs update was on friday the GPS addition. Was this version running?
I will look up the possible problems with my fater on the weekend, but we are not sure that we can help. There are just too many environmental dependencies, we can't really reproduce.

**Comment by NVB9ALT** [comment]
Last time it was running properly was 7-9 days ago, I think.
I have no idea.
No, as this was before the GPS update.

I'll try and log all the different types of malfunctions, and once it's done I'll show you.

**Comment by kolos26** [comment]
Thanks, that might be useful. Can you run the legacy or the beta version through the proxy?

**Comment by NVB9ALT** [comment]
No, most of the time the proxy website (geofs-proxy.coveljulian.repl.co) doesn't even load lol.

**Comment by NVB9ALT** [comment]
I heard it wasn't fully up to date, so I deleted the repl and I'm gonna try and get a new deployment from GitHub.

**Comment by kolos26** [comment]
That may help, for me it wasn't even load

**Comment by NVB9ALT** [comment]
Actually it won't run at all, something about ToS violations or some such.

**Comment by NVB9ALT** [comment]
Do you know a way to run arbitrary JavaScript in an iframe? My idea is to create a website with an iframe of proxied GeoFS, then append the scripts for a console and the Realism Pack to that iframe.

**Comment by kolos26** [comment]
try this:
`document.getElementById("idOfIframe").contentDocument.appendChild(_here comes a <script> tag with the code_)`
[https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/contentDocument](https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/contentDocument)
if this don't work try contentWindow under that

And honestly, I have never tried it, I have just googled it up :)

**Comment by NVB9ALT** [comment]
Oh boy I sure do love the cross-origin policy.

**Comment by kolos26** [comment]
> Oh boy I sure do love the cross-origin policy.

1. I created a free Netlify account [https://www.netlify.com/](https://www.netlify.com/)
2. Choosed a Hugo quickstart template
3. Copied all file to the `/static` folder
4. In the `/static` folder I created a file called `_headers`
This comes into `_headers`:
``` 
/*
Access-Control-Allow-Origin: *
```
It worked for me! Good luck!

**Comment by NVB9ALT** [comment]
What files go in /static? The html/js of your website, or do I also need to put all GeoFS files in there?

**Comment by kolos26** [comment]
Html and js of the iframe page

**Comment by NVB9ALT** [comment]
Couldn't find an editor on Netlify itself; and there's no "static" folder in the GitHub repo that Netlify is deploying.

**Comment by kolos26** [comment]
You need to create it yourself

**Comment by NVB9ALT** [comment]
Well I added a /static folder to the repo, and put my HTML/JS files in there, but it didn't change anything on the deployed site. Do I need to delete themes/netlify-basic?

**Comment by kolos26** [comment]
Have you added `_headers` too?

**Comment by NVB9ALT** [comment]
[Screen recording 2023-04-10 9.39.50 AM.webm](https://user-images.githubusercontent.com/102188956/230948370-6b3d9448-4cad-49d8-8413-0f5a0ea83358.webm)

**Comment by kolos26** [comment]
> > Oh boy I sure do love the cross-origin policy.
> 
> 1. I created a free Netlify account https://www.netlify.com/
> 2. Choosed a Hugo quickstart template
> 3. Copied all file to the `/static` folder
> 4. In the `/static` folder I created a file called `_headers`
>    This comes into `_headers`:
> 
> ```
> /*
> Access-Control-Allow-Origin: *
> ```
> 
> It worked for me! Good luck!

**Comment by kolos26** [comment]
And you need to open
On the deploynent the /static/index.htnl

**Comment by NVB9ALT** [comment]
Adding _headers didn't change anything
![image](https://user-images.githubusercontent.com/102188956/230949110-86501ef5-b4c1-4fde-868b-afad7af256f3.png)

**Comment by NVB9ALT** [comment]
Set base directory to /static in build settings and it made the deploy fail.

**Comment by kolos26** [comment]
Do that setting back, it must work in a different way.
https://hypertabs.cc/uv/service/hvtrs8%2F-wuw%2Cggo%2Ffq.aoo%2Feemfq.rhr - this link is broken
https://geofs-iframe-ex.netlify.app/static/index.html - open this link after bulid

**Comment by NVB9ALT** [comment]
The first link works fine for me; you should probably load https://hypertabs.cc before using that link to make sure the proxy runs.

**Comment by kolos26** [comment]
it works, but that link throws 404

**Comment by NVB9ALT** [comment]
then just type in https://geo-fs.com/geofs.php into the url bar of the hypertab. might be interesting to see if the links are different between two users doing the same thing.

**Comment by NVB9ALT** [comment]
and the second link throws 404 not found, i did re-deploy the site but it didn't fix it

**Comment by kolos26** [comment]
I will think on this tommorow, sorry but my brain is off now. Bye Kolos

**Comment by NVB9ALT** [comment]
You should honestly just make a video demonstrating how you did it earlier

> > Oh boy I sure do love the cross-origin policy.
> 
> 1. I created a free Netlify account https://www.netlify.com/
> 2. Choosed a Hugo quickstart template
> 3. Copied all file to the `/static` folder
> 4. In the `/static` folder I created a file called `_headers`
>    This comes into `_headers`:
> 
> ```
> /*
> Access-Control-Allow-Origin: *
> ```
> 
> It worked for me! Good luck!

**Comment by kolos26** [comment]
I will make a video of course, but, what about if you simply rename index.html to something else?

**Comment by NVB9ALT** [comment]
Renamed to main.html, didn't work
